HotelsGUI/MainWindow.xaml.cs

```diff
@@ -177,16 +177,35 @@ private void RefreshGUI()
             OutputTextBlock.Text = string.Empty;
         }
 
-        private SavedPreference GetSavedPreferenceFromInputs() =>
-            new SavedPreference
+        private SavedPreference GetSavedPreferenceFromInputs()
+        {
+            return (SavedPreference)(new SavedPreference()
```

Comment:
tego troche nie do konca lapie ale spoko :P

Reply:
Metoda zwraca SavedPreference ale wszystkie metody z bazowej UserPreference zwracaja typ UserPreference. Więc wywołując łańcuchowo metody With..() dostanę finalnie obiekt typu UserPreference a potrzebuje zwrócić SavedPreference, dlatego castuje ale być może można to jakoś upięknić